### PR TITLE
Refactor/operations

### PR DESCRIPTION
**Pull request by abeforgit**
This is an effort to make the model.change flow more intuitive
The existing logic hasn't changed, it's just been renamed to make their batched behavior more clear, and a non-batched version has been added 

For the moment, we have no real use for the batched version so it's mostly getting in the way.
depends on #87

### CODE REVIEW

**Comment by abeforgit** [comment]
good thing its a draft, immediate obvious mistake spotted

File: addon/model/model.ts [hunk 1]
```diff
@@ -144,6 +144,11 @@ export default class Model {
     modelNode.remove();
   }
 
+  /**
```
Comment (nvdk) [inline addon/model/model.ts:147]: huuraah for docs!

File: addon/commands/insert-xml-command.ts [hunk 2]
```diff
@@ -22,7 +22,8 @@ export default class InsertXmlCommand extends Command {
     const range = selection.lastRange;
     const parsed = parseXml(xml);
     this.model.change(mutator => {
-      mutator.insertNodes(range, parsed.root);
+      const newRange = mutator.insertNodes(range, parsed.root);
+      mutator.selectRange(newRange);
```
Comment (nvdk) [inline addon/commands/insert-xml-command.ts:26]: for my understanding, it's now expected that a command explicitly selects something or do we still (also) provide a "best guess" selection?

**Comment by nvdk** [comment]
hmm accidentally pushed to this branch, apologies

File: addon/commands/insert-xml-command.ts (hunk 2 above)
Comment (abeforgit) [reply addon/commands/insert-xml-command.ts:26]: yes, thats what insertNodes returns
but the command has to decide to use it indeed, it is not automatically applied

**Review by nvdk — APPROVED** [review]
looks good to me!

**Comment by nvdk** [comment]
> looks good to me!

woops looks like this introduces a linting error :)

**Pull request merged**